generic3g/actions/AccumulatorAction.F90

```diff
@@ -0,0 +1,201 @@
+#include "MAPL_Generic.h"
+module mapl3g_AccumulatorAction
+   use mapl3g_ExtensionAction
+   use MAPL_InternalConstantsMod, only: MAPL_UNDEFINED_REAL, MAPL_UNDEFINED_REAL64
+   use MAPL_FieldUtilities, only: FieldSet 
+   use MAPL_FieldPointerUtilities
+   use MAPL_ExceptionHandling
+   use ESMF
+   implicit none
+   private
+   public :: AccumulatorAction
+
+   type, extends(ExtensionAction) :: AccumulatorAction
+      logical :: update_calculated = .FALSE.
+      type(ESMF_Field), allocatable, public :: accumulation_field
+      type(ESMF_Field), allocatable, public :: result_field
+      real(kind=ESMF_KIND_R4) :: UNDEF_VALUE_R4 = MAPL_UNDEFINED_REAL
+      real(kind=ESMF_KIND_R4) :: CLEAR_VALUE_R4 = 0.0_ESMF_KIND_R4
+   contains
+      ! Implementations of deferred procedures
+      procedure :: invalidate
+      procedure :: initialize
+      procedure :: update
+      ! Helpers
+      procedure :: accumulate
+      procedure :: initialized
+      procedure :: clear_accumulator
+      procedure :: clear_fields
+      procedure :: accumulate_R4
+      procedure :: calculate_result
+   end type AccumulatorAction
+
+contains
+
+   logical function initialized(this) result(lval)
+      class(AccumulatorAction), intent(in) :: this
+
+      lval = allocated(this%accumulation_field)
+
+   end function initialized
+
+   subroutine clear_accumulator(this, rc)
+      class(AccumulatorAction), intent(inout) :: this
+      integer, optional, intent(out) :: rc
+      integer :: status
+   
+      call this%clear_fields(_RC)
+      _RETURN(_SUCCESS)
+
+   end subroutine clear_accumulator
```

Comment:
This method appears to be unnecessary.   Just transfers to `clear_fields`

Reply:
Refactored to accomodate the variation for clear_accumulator for subtypes.